memory_report.cpp

```diff
@@ -140,6 +143,26 @@ void report_memory_consumption(
    MPI_Comm_rank(interComm, &interRank);
    MPI_Comm_size(interComm, &nNodes);
 
+#if __GNUC__ >= 13
+   // string formatting
+   constexpr std::string_view formatString = "(MEM) tstep {} t {:.3g} {:<21} (GiB/node; avg, min, max, sum): {:<8.3g} {:<8.3g} {:<8.3g} {:<8.3g} on {} nodes\n";
+#endif
+
+   // Report /proc/meminfo memory consumption first so we then get resident and HWM from Papi below, easier to compare by eye in logfile in this order.
+   double mem_proc_free = (double)get_node_free_memory();
+   double total_mem_proc = 0;
+   double min_free,max_free;
+   const int numberOfParameters = 1;
+   MPI_Reduce( &mem_proc_free, &total_mem_proc, numberOfParameters, MPI_DOUBLE, MPI_SUM, MASTER_RANK, MPI_COMM_WORLD );
```

Comment:
This is wrong. If running with n mpi tasks per node, it sums the same node n times.
(Just tested on 1 node with 4 tasks. With 8 GB free, it reports ```min 8, max 8, sum 32```).

I guess this needs to be performed on the interComm?

Reply:
Also, the avg calculation then needs to divide by nNodes, not nProcs.